src/hotspot/share/runtime/threads.cpp

```diff
@@ -1366,10 +1366,24 @@ void Threads::print_on(outputStream* st, bool print_stacks,
   char buf[32];
   st->print_raw_cr(os::local_time_string(buf, sizeof(buf)));
 
-  st->print_cr("Full thread dump %s (%s %s):",
+  st->print_cr("Full thread dump %s (%s %s)",
                VM_Version::vm_name(),
                VM_Version::vm_release(),
                VM_Version::vm_info_string());
+  JDK_Version::current().to_string(buf, sizeof(buf));
+  const char* runtime_name = JDK_Version::runtime_name() != nullptr ?
+                             JDK_Version::runtime_name() : "";
+  const char* runtime_version = JDK_Version::runtime_version() != nullptr ?
+                                JDK_Version::runtime_version() : "";
+  const char* vendor_version = JDK_Version::runtime_vendor_version() != nullptr ?
+                               JDK_Version::runtime_vendor_version() : "";
+  const char* jdk_debug_level = VM_Version::printable_jdk_debug_level() != nullptr ?
+                                VM_Version::printable_jdk_debug_level() : "";
+
+  st->print_cr("                 JRE version: %s%s%s (%s) (%sbuild %s)", runtime_name,
```

Comment:
The move to modules in JDK 9 blurred the historical distinction between what we used to know as the JDK and JRE. So we don't use "JRE" anymore. I think it would be okay to say JDK or runtime version here.

Reply:
related to https://bugs.openjdk.org/browse/JDK-8330123

Reply:
> We probably need to file a general fix up for this as we still report "JRE" in -Xinternalversion, as well as the version string in the hs_err report.

I did a quick search of src/hotspot and these seems to the only two. Yes, would be good to update both.